src/CommunityToolkit.Maui/Extensions/AppThemeExtension.shared.cs

```diff
@@ -0,0 +1,91 @@
+﻿namespace CommunityToolkit.Maui.Extensions;
+
+/// <summary>
+/// A XAML markup extension that enables using <see cref="AppThemeColor"/> and <see cref="AppThemeResource"/> from XAML.
+/// </summary>
+[ContentProperty(nameof(Key))]
+public sealed class AppThemeExtension : IMarkupExtension<BindingBase>
+{
+	/// <summary>
+	/// Gets or sets the key that is used to access the <see cref="AppThemeColor"/> or <see cref="AppThemeResource"/> from the <see cref="ResourceDictionary"/>.
+	/// </summary>
+	public string? Key { get; set; }
+
+	/// <inheritdoc/>
+	/// <exception cref="ArgumentNullException">Thrown if <paramref name="serviceProvider"/> is <see langword="null"/>.</exception>
+	/// <exception cref="XamlParseException">Thrown if <see cref="Key"/> is <see langword="null"/> or if a resource with <see cref="Key"/> cannot be found.</exception>
+	/// <exception cref="ArgumentException">Thrown if <paramref name="serviceProvider"/> does no implement <see cref="IProvideParentValues"/>.</exception>
+	public BindingBase ProvideValue(IServiceProvider serviceProvider)
+	{
+		if (serviceProvider is null)
+		{
+			throw new ArgumentNullException(nameof(serviceProvider));
+		}
+
+		if (Key is null)
+		{
+			throw new XamlParseException("You must specify a key for {AppThemeExtension} that specifies the AppTheme resource to use", serviceProvider);
```

Comment:
Does sourceProvider a valid parameter?

Reply:
Yup, it's a valid parameter for `XamlParseException`